### PR TITLE
Machine Keys, Orleans\Fusion Streams, Registration and User Argreements 

### PR DESCRIPTION
**Pull request by 0xF6**
[#33](https://github.com/argon-chat/argon/issues/33)
[#43](https://github.com/argon-chat/argon/issues/43)
[#44](https://github.com/argon-chat/argon/issues/44)
[#45](https://github.com/argon-chat/argon/issues/45)

## Release Notes

- **New Features**
  - Introduced `AuthorizationController` for user authorization and registration.
  - Added `UserAgreements` management capabilities in the database.
  - Implemented WebSocket server functionality for RPC communications.

- **Enhancements**
  - Updated various package dependencies for improved performance and security.
  - Enhanced logging for Entity Framework Core operations.

- **Bug Fixes**
  - Addressed issues related to user session management and token validation.
  - Improved error handling in user authorization processes.

- **Refactor**
  - Restructured user and server interaction services for better maintainability.
  - Cleaned up and organized migration files for database schema updates.

- **Documentation**
  - Updated internal documentation to reflect new features and changes.

### CODE REVIEW

File: src/Argon.Api/Entities/Channel.cs [hunk 1]
```diff
@@ -22,9 +16,6 @@ public sealed record Channel
     public ChannelType ChannelType { get; set; } = ChannelType.Text;
```
Comment (urumo) [inline src/Argon.Api/Entities/Channel.cs:16]: если ты удалил ChannelType энум то почему тут оставил?

File: src/Argon.Api/Entities/User.cs [hunk 2]
```diff
@@ -21,9 +23,20 @@ public sealed record User
...
     public List<UsersToServerRelation> UsersToServerRelations { get; set; } = new();
+}
+
+
+public record UserAgreements
+{
+    [Key]
+    public Guid     Id                        { get; set; } = Guid.NewGuid();
+    public bool     AllowedSendOptionalEmails { get; set; }
+    public bool     AgreeTOS                  { get; set; }
```
Comment (urumo) [inline src/Argon.Api/Entities/User.cs:34]: дефолтно поставь false для гомогенности кода

File: src/Argon.Api/Entities/Channel.cs (hunk 1 above)
Comment (0xF6) [reply src/Argon.Api/Entities/Channel.cs]: ты инвалид ебанный внимательнее смотри

File: src/Argon.Api/Entities/User.cs (hunk 2 above)
Comment (0xF6) [reply src/Argon.Api/Entities/User.cs:34]: я тебе сейчас гомосексуальнокодоности втащу 10 миллилитров

**Review by urumo — APPROVED** [review]

**Pull request merged**